src/components/documentComponents/documentLinkSearch/DocumentLinkList.js

```diff
@@ -0,0 +1,73 @@
+import { request } from "../../../utils.js";
+import DocumentLink from "./DocumentLink.js";
+
+export default function DocumentLinkList({ $parent, $target, initialState, onClose }) {
+  const $documentLinkList = document.createElement("ul");
+  $documentLinkList.className = "document-link-list";
+
+  $target.appendChild($documentLinkList);
+
+  this.state = initialState;
+
+  this.componentDidMount = async () => {
+    const documents = await request("");
+    const documentLinks = documents.map(({ id, title, documents }) => ({
+      id,
+      title,
+      documents: documents.map(({ id, title, documents }) => ({
+        id,
+        title,
+        documents: documents.map(({ id }) => ({ parentId: id })),
+      })),
+    }));
+    this.setState({ documentLinks });
+  };
```

Comment:
fetch 하는 함수로 따로 분리하는 것도 좋을 것 같습니다. 아래와 같이요. 그리고, request 함수에서 에러를 던지고 있는데, 그 에러를 잡아주는 부분이 빠진 것 같습니다.
```suggestion
 const fetchDocuments = async () => {
    try {
      const documents = await request("");
    return documents.map(({ id, title, documents }) => ({
      id,
      title,
      documents: documents.map(({ id, title, documents }) => ({
        id,
        title,
        documents: documents.map(({ id }) => ({ parentId: id })),
      })),
    }));
    } catch (error) {
      console.error(error);
    }
  }

  this.componentDidMount = async () => {
    const documentLinks = await fetchDocuments();
    this.setState({ documentLinks });
  };
```

Reply:
아 그렇네요 api 호출인데 try catch를 빼먹다니